### PR TITLE
Fix #469 - Remove sensitive header on TRACE method

### LINKED ISSUE #469: AS per RFC 9110 (and earlier) HTTp TRACE response MUST not include sensitive headers
https://www.rfc-editor.org/rfc/rfc9110.html#name-trace

The requirement was added in RFC 7231. It is not present in RFC 2616.

### PR DESCRIPTION
**Pull request by dsandrade0**
This commit

### CODE REVIEW

File: api/src/main/java/jakarta/servlet/http/HttpServlet.java
Comment (markt-asf) [inline api/src/main/java/jakarta/servlet/http/HttpServlet.java:515]: Does this really need a separate method?

**Review by markt-asf — CHANGES_REQUESTED** [review]
The code formatter has been updated so I am afraid you'll need to reformat the code and update the PR.

It would be helpful if you included a changes entry for this in the spec document.

File: api/src/main/java/jakarta/servlet/http/HttpServlet.java
Comment (dsandrade0) [reply api/src/main/java/jakarta/servlet/http/HttpServlet.java:515]: In this [RFC](https://www.rfc-editor.org/rfc/rfc9110.html#name-trace) don't have the list of sensitive headers. Then include on this method because Java method will be seen by JIT to compiler separated of #doTrace and will be better performance.

Comment (markt-asf) [reply api/src/main/java/jakarta/servlet/http/HttpServlet.java:515]: All JIT is likely to do is in-line the method - which is what I am suggesting you do anyway. It isn't as if the separate method adds to the readability of the code.

File: api/src/main/java/jakarta/servlet/http/HttpServlet.java
Comment (markt-asf) [inline api/src/main/java/jakarta/servlet/http/HttpServlet.java:83]: Why these two methods? `Authorization` I understand but not `Accept`. And why not `Cookie`?

File: api/src/main/java/jakarta/servlet/http/HttpServlet.java
Comment (gregw) [reply api/src/main/java/jakarta/servlet/http/HttpServlet.java:515]: Either in line it, or make it protected so that an extender can change what headers are considered safe.

File: api/src/main/java/jakarta/servlet/http/HttpServlet.java
Comment (gregw) [reply api/src/main/java/jakarta/servlet/http/HttpServlet.java:83]: I agree that `Cookie` is a sensitive header, whilst no so much `Accept`.

I also think that any proxy headers (`X-Forwarded-*`, `Forwarded`,  `Proxy-Authorization`) should be protected so that trace can't be used to probe the internals of a data centre.

File: api/src/main/java/jakarta/servlet/http/HttpServlet.java
Comment (gregw) [inline api/src/main/java/jakarta/servlet/http/HttpServlet.java:496]: Should the headers be skipped, or should just there values be obscured?  RFC kind of says both.

**Review by gregw — CHANGES_REQUESTED** [review]

File: api/src/main/java/jakarta/servlet/http/HttpServlet.java
Comment (dsandrade0) [reply api/src/main/java/jakarta/servlet/http/HttpServlet.java:496]: Maybe obscure can show that theses headers exists. Should the response knows theses headers are present?

File: api/src/main/java/jakarta/servlet/http/HttpServlet.java
Comment (dsandrade0) [reply api/src/main/java/jakarta/servlet/http/HttpServlet.java:83]: Then I will remove the class field HEADER_SENSITIVE and move all implementation to isSensitiveHeader method

File: api/src/main/java/jakarta/servlet/http/HttpServlet.java
Comment (gregw) [reply api/src/main/java/jakarta/servlet/http/HttpServlet.java:496]: I think leave it as skipping.  If somebody wants to see more then they can implement doTrace themselves.

**Comment by markt-asf** [comment]
The review comments have been mostly addressed but I am going to apply this manually so I can make a few tweaks including:
- Adding the change to the change log section in the specification
- Re-wording the explanation of sensitive headers added to the spec
- Generate the default list of headers to exclude once and re-use it
- Make header comparison case insensitive
- Add Javadoc to `isSensitiveHeader()`
- Rename test methods
- Refactor test slightly to avoid NPE warning

**Comment by markt-asf** [comment]
Greg's comments have been addressed so I am going to dismiss that review as I think not doing so will block the merge of the PR (even doing it manually). If I have missed anything, I'm happy to take an action to get it fixed.

**Comment by markt-asf** [comment]
Tx. Merged via #519

**Pull request closed without merging**
